figshare/figshare.py

```diff
@@ -290,3 +293,37 @@ def get_file_details(self, article_id, file_id):
         response = issue_request('GET', url,
                                  headers=self.get_headers(token=self.token))
         return response
+
+    def retrieve_files_from_article(self, article_id, outpath=''):
+        """ Retrieve files and save them locally.
+
+        By default, files will be stored in the current working directory
+        under a folder called figshare_<article_id> by default.
+        Specify <outpath> for: <outpath>/figshare_<article_id>
+
+        Parameters
+        ----------
+        article_id : str or int
+            Figshare article ID
+
+        """
+
+        # Handles missing trailing forward slash
+        if outpath != '':
+            if outpath[-1] != '/':
+                outpath += '/'
+
+        # Get list of files
+        file_list = self.list_files(article_id)
+
+        n_files = len(file_list)
+        if n_files != 0:
+            dir0 = outpath+"figshare_{0}/".format(article_id)
+            os.makedirs(dir0, exist_ok=True) # This might require Python >=3.2
+
+            for ii in range(n_files):
+               t_name = dir0 + file_list[ii]['name']
```

Comment:
Will name ever contain characters not supported by the filesystem?

Reply:
Interesting point.  I think we would need to find such an example to better test.

From my interactions with Figshare articles, I have not seen special characters.  I'm not sure if Figshare even allows upload with special characters in the filename.

The only mention of special characters I found in Figshare doc is this (for their search feature):
https://docs.figshare.com/#search_advanced_search

Reply:
I'm fine with ignoring this problem for now. People shouldn't be uploaded files with unsafe names... although some probably do.

Reply:
We can open an enhancement issue for this.  If there's a way to identify special characters in a string, we can perhaps default to writing files as file1... file2..., etc.